subsys/net/lib/lwm2m/lwm2m_obj_firmware.c

```diff
@@ -83,7 +158,16 @@ static int package_uri_write_cb(u16_t obj_inst_id,
 {
 	SYS_LOG_DBG("PACKAGE_URI WRITE: %s", package_uri);
 #ifdef CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT
-	lwm2m_firmware_start_transfer(package_uri);
+	u8_t state = lwm2m_firmware_get_update_state();
+	if (STATE_IDLE == state) {
+		lwm2m_firmware_set_update_result(RESULT_DEFAULT);
+		lwm2m_firmware_start_transfer(package_uri);
+	} else if (STATE_DOWNLOADED == state) {
+		if (data_len == 0) {
+			lwm2m_firmware_set_update_state(STATE_IDLE);
+			/* TODO: clean-up downloaded package? */
+		}
+	}
```

Comment:
I'm afraid that I did not catch your point.
What kind of checks do I have to add to package_uri_write_cb?